radixdlt/src/main/java/com/radixdlt/consensus/EventCoordinator.java

```diff
@@ -135,21 +168,15 @@ public void processProposal(Vertex proposedVertex) {
 			return;
 		}
 
-		final VoteResult voteResult;
+		mempool.removeCommittedAtom(atom.getAID());
+
 		try {
-			voteResult = safetyRules.voteFor(proposedVertex);
-			final Vote vote = voteResult.getVote();
+			final Vote vote = safetyRules.voteFor(proposedVertex);
 			networkSender.sendVote(vote);
-			voteResult.getCommittedVertexId()
-				.ifPresent(vertexId -> {
-					log.info("Committed vertex " + vertexId);
-
-					final Vertex vertex = vertexStore.commitVertex(vertexId);
-					final Atom committedAtom = vertex.getAtom();
-					mempool.removeCommittedAtom(committedAtom.getAID());
-				});
```

Comment:
Why is the logic above removed?

Reply:
Wasn't removed, was just moved to another place - see lines 117:121

Reply:
Don't think this should be removed from here though. Processing a new proposal also means processing unseen QCs.

Reply:
You're totally right, that should be at the start though